### PR TITLE
Add IAM sample for restricting access to BQ datasets

### LINKED ISSUE #314: Add samples to check for BigQuery datasets exposed to gmail & googlegroups
Forseti BigQuery scanner checks for datasets exposed to users with *@gmail.com and *@googlegroups.com email addresses. Add a sample to check for these and include the forseti annotation.

### PR DESCRIPTION
**Pull request by gkowalski-google**
Add samples restricting access to BQ datasets for gmail.com and googlegroups.com email addresses. Add unit tests for these samples, and update existing unit tests for this template to new format.

Resolves #314

### CODE REVIEW

File: samples/iam_restrict_gmail_bigquery_dataset.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+apiVersion: constraints.gatekeeper.sh/v1alpha1
+kind: GCPIAMAllowedBindingsConstraintV1
+metadata:
+  name: blacklist_gmail_bigquery_dataset
```
Comment (morgante) [inline samples/iam_restrict_gmail_bigquery_dataset.yaml]: Names should use hyphens, not underscores, for forwards-compatibility.

File: validator/iam_allowed_bindings_test.rego [hunk 2]
```diff
@@ -16,82 +16,61 @@
...
-
-	violation := found_violations[_]
+# Test blacklist public
+test_blacklist_role_violation_count {
+	test_utils.check_test_violations_count(fixture_assets, [fixture_constraints.iam_allowed_bindings_blacklist_public], template_name, 2)
 }
 
-test_blacklist_project_violations {
-	count(blacklist_project_violations) = 1
+# Test blacklist role
```
Comment (morgante) [inline validator/iam_allowed_bindings_test.rego]: Duplicate test from line 26?

File: samples/iam_restrict_gmail_bigquery_dataset.yaml (hunk 1 above)
Comment (gkowalski-google) [reply samples/iam_restrict_gmail_bigquery_dataset.yaml]: Updated.

File: validator/iam_allowed_bindings_test.rego (hunk 2 above)
Comment (gkowalski-google) [reply validator/iam_allowed_bindings_test.rego]: Thanks for catching, the methods were duplicated but the tests were not. Update the names to reflect what is actually being tested.

**Comment by gkowalski-google** [comment]
@morgante Thanks for the review comments, I have addressed, can you please review again?

**Pull request merged**